xds/src/main/java/io/grpc/xds/WeightedRoundRobinLoadBalancer.java

```diff
@@ -266,25 +265,26 @@ private void updateWeight() {
           weightedChannelCount++;
         }
       }
-      if (weightedChannelCount < 2) {
-        rrMode = true;
-        return;
-      }
       EdfScheduler scheduler = new EdfScheduler(list.size(), random);
-      avgWeight /= 1.0 * weightedChannelCount;
+      if (weightedChannelCount >= 2) {
+        avgWeight /= 1.0 * weightedChannelCount;
+      }
       for (int i = 0; i < list.size(); i++) {
         WrrSubchannel subchannel = (WrrSubchannel) list.get(i);
-        double newWeight = subchannel.getWeight();
-        scheduler.add(i, newWeight > 0 ? newWeight : avgWeight);
+        if (weightedChannelCount >= 2) {
```

Comment:
If you assign avgWeight to 1 before the for loop, only for the `weightedChannelCount == 0` case, then you don't need this condition.

```java
      if (weightedChannelCount >= 1) {
        avgWeight /= 1.0 * weightedChannelCount;
      } else {
        avgWeight = 1;
      }
```